src/storage/src/manifest/region.rs

```diff
@@ -334,6 +334,13 @@ mod tests {
             .await
             .is_ok());
 
+        assert!(manifest
```

Comment:
nitpick: Could we add a new test (`test_region_manifest_truncate`)? This test case is too complicated now 😢 

The test may write a RegionChange and a RegionTruncate and scan it.